fix(NXP-30164): code review

jcarsique opened the pull request:
Code review: fixes, upgrades, cleanup, pipeline & preview improvements
(aligned with nuxeo-ai CI code)

jcarsique committed:
fix(NXP-30164): code review

chores: align with Nuxeo 11.4
chores: use nuxeo-web-ui 3.1.0-rc.22 from NOS PREPROD
chores: preview upgrade to Nuxeo chart ~1.2.1
chores: upgrade version stream
cleanup: debug code
refactor: use waitForNuxeo(String, String, Closure) instead of waitForNuxeo(String, String)
refactor: GitHub statuses renamed: build/maven -> maven/build, build/docker -> docker/build
fix: SCM_REF use long SHA1
ci: remove withMaven()
ci: preview deployment improved and not blocking the pipeline continuation
ci: node affinity, pod toleration
ci: use jenkins-ai-nuxeo11 pod template
fix: do not include the release tag in master history
fix: release version eval

nuxeojenkins commented:
View issue in JIRA: [NXP-30164](https://jira.nuxeo.com/browse/NXP-30164): Release nuxeo-vision package for 11.4 in Nuxeo Marketplace

jcarsique committed:
fix(NXP-30164): deprecate addon

nuxeo-vision is deprecated, replaced by nuxeo-ai-gcp-core addon (part of nuxeo-ai)
